src/wdio/services/compare/TerraCompare.js

```diff
@@ -0,0 +1,54 @@
+import BaseCompare from 'wdio-visual-regression-service/lib/methods/BaseCompare';
+import resemble from 'node-resemble-js';
+import get from 'lodash.get';
+
+export default class TerraCompare extends BaseCompare {
+  constructor(screenshotProcessor) {
+    super();
+    this.screenshotProcessor = screenshotProcessor;
+  }
+
+  async processScreenshot(context, base64Screenshot) {
+    const referenceScreenshotData = await this.screenshotProcessor.getReferenceScreenshotData(context);
+    const latestScreenshotData = Buffer.from(base64Screenshot, 'base64');
+    await this.screenshotProcessor.saveLatestScreenshotData(context, latestScreenshotData);
+    if (referenceScreenshotData) {
+      const comparisonData = await TerraCompare.compareImages(referenceScreenshotData, latestScreenshotData);
+      const { isSameDimensions } = comparisonData;
+      const misMatchPercentage = Number(comparisonData.misMatchPercentage);
+      const misMatchTolerance = get(context, 'options.misMatchTolerance', 0.01);
+
```

Comment:
nit: since we are calling back to the BaseCompare method, can we defined
```diff
if (referenceScreenshotData) {
  // stuff
+ const isWithinMisMatchTolerance = misMatchPercentage <= misMatchTolerance

+ if (!isWithinMisMatchTolerance) {
- if (misMatchPercentage > misMatchTolerance) {
 // stuff
 // don't reporter to do outside the if block
-     return this.createResultReport(misMatchPercentage, false, isSameDimensions);
 }

- return this.createResultReport(misMatchPercentage, true, isSameDimensions);
+ return this.createResultReport(misMatchPercentage, isWithinMisMatchTolerance, isSameDimensions);
` ``